### PR TITLE
This is a Microverse project on positioning and floating elements.

### PR DESCRIPTION
**Pull request by Honda99**
**About the Project**
This project is about building a news article page from New York Times magazine

Title of the article is Space Ripples Reveal Big Bang’s Smoking Gun and the link:

[](https://www.nytimes.com/2014/03/18/science/space/detection-of-waves-in-space-buttresses-landmark-theory-of-big-bang.html?_r=0)

The website has a header:
With menu icon, NYT logo, and two buttons to subscribe and log in.
Have a news article in the main section. News story contains information like:
Publisher name
Published date.
Social icons to share the news.
Bookmark and comment on icons.
News story.
Button to view comments.
Editor's pick on the right side to show specific news from the editor.
Grid section with more news.
Popular news on the right side.
Has a footer with
Useful links like; News, Opinion, More, Arts, Living, and Subscribe.
Copy the right information and other site-specific links.

### CODE REVIEW

File: styles.css [hunk 1]
```diff
@@ -1,456 +1,457 @@
...
 #homepage {
-  display: table-cell;
-  width: 30%;
-  float: right;
-  color: blue;
+    display: table-cell;
+    width: 30%;
+    float: right;
+    color: blue;
 }
```
Comment (uryelah) [inline styles.css]: - [ ] Using id selectors in CSS is a bad practice that can make styling your code harder and buggier as its complexity increases. Please change those id selectors for class selectors and [read this article](https://dev.to/clairecodes/reasons-not-to-use-ids-in-css-4ni4) on reasons why using ids in CSS is not the best idea. (They will be very useful in JavaScript though ;-))

File: index.html [hunk 2]
```diff
@@ -72,7 +72,7 @@ <h4>By Dennis Overbye</h4>
                         smoking-gun evidence of inflation, proof, Dr. Kovac and his colleagues say, that Dr. Guth was correct. Inflation has been the workhorse of cosmology for 35 years, though many, including Dr. Guth, wondered whether it could ever
                         be proved.
                         <br><br> If corroborated, Dr. Kovac’s work will stand as a landmark in science comparable to the recent discovery of dark energy pushing the universe apart, or of the Big Bang itself. It would open vast realms of time and space
-                        and energy to science and speculation.
+                        and energy to science.
                         <br><br> Confirming inflation would mean that the universe we see, extending 14 billion light-years in space with its hundreds of billions of galaxies, is only an infinitesimal patch in a larger cosmos whose extent, architecture
                         and fate are unknowable. Moreover, beyond our own universe there might be an endless number of other universes bubbling into frothy eternity, like a pot of pasta water boiling over.
                     </p>
```
Comment (uryelah) [inline index.html]: How the page currently looks when opening it locally(after fixing the image paths):
https://ibb.co/XF9ptJJ

**Review by uryelah — CHANGES_REQUESTED** [review]
Hi **Boaz Aduda & Yosef Welegergs**, I'm Sarah your reviewer for this PR 👩‍🚀 

Good job kick-starting this project and delivering it after all the milestones!

I'll ask you to take some extra time to carefully re-read the milestone 5 and 6, and then come back and read this review carefully. 

Make sure to be very attentive of the requirements for this project and give your best go at fixing the current issues with it.

Rock on! 🎸 👨‍🎤 

# Required Changes 🏗️ 🐛 💡

-----

 **Kindly fix the pointed out items and comments below before requesting the next code review**

- [ ]  Please make sure to run your linters locally before submitting the project for review and making sure there are no errors or warnings. Currently, some of your linter checks are not passing. 
Make sure to fix them, that will improve the quality of your code dramatically, read more about [setting up and running linters here](https://github.com/microverseinc/linters-config/tree/master/html-css).

- [ ] Milestone 6 of this project requires you to deploy a live version of your project and include it's link in the README file. Kindly follow the recommendations as seen at [the Pathwright instructions](https://microverse.pathwright.com/library/new-technical-curriculum/177956/path/step/104230602/).
Tip: You can deploy your HTMl projects in multiple ways but the easiest ones would be using [GitHack](https://raw.githack.com/) or [GitHub pages](https://guides.github.com/features/pages/).

- [ ] [OPTIONAL] You can just use the [embed youtube tag/link to display the video](https://support.google.com/youtube/answer/171780?hl=en) in the page, no need to download the whole video, that takes a lot of space.

- [ ] Format your HTML better. Use only 2 spaces for the indentation instead of 4 and only 1 line at most between two tags.

- [ ] Never leave `<a>` tags without and `href` attribute or with an empty `href=""`.  that's a bad practice semantically,
If the link is not linking to any specific href just use `href="#"` so it links to the current page.

- [ ] Use the complete relative path when linking images, videos, CSS files, or any asset you have. Otherwise, your page might not display the assets on GitHack or when opening the page in the browser using the relative HTML path.
```
Good: 
'./images/file_name'

Bad:
'/images/file_name'
```
That dot before the first slash, `./`, indicates the current location of your file in the folder, make sure to include it at the start of all your image links, CSS links, and video links.

- [ ] next projects name your branch better, it should be named after the feature you just built, in this case, the full NY Times page and not the submission status would be the feature to name the branch.

- [ ] Always add an `alt` property to your `<image>` tags. That's for the sake of accessibility so that people that use screen readers to browse the web still are able to understand what the image content is. Read [this article ](https://www.w3schools.com/tags/att_img_alt.asp)for more.

- [ ] Never use inline styles inside tags such as:
 `<hr style="width: 30%; height: 1px; border: none; background-color: black;">`
That is another CSS bad practice that can make future style debugging harder and make the growth of a code base unsustainable. 
Remember to follow separation of concerns: keep all styles in CSS fiels and all the markup in HTML files.

----

**Styling**:

- [ ] Change the color of the links and buttons so they match the ones in the original page.
Tip, if you use Firefox you can use the [built in color picker tool](https://developer.mozilla.org/en-US/docs/Tools/Eyedropper) in its developer tools.
Remember, when you work professionally in a project the designer will expect you to know how to use the specific colors present in the website layout.

- [ ] Work more on the Footer section to make sure it has the same spacings and styles of the original page.

- [ ] Don't use screenshots for the logo, search for the ny times logo in google images, ideally finding an image with the .png extension and a transparent background or an svg logo. [(Link about using SVG and PNG images)](https://weblium.com/blog/svg-vs-png-what-is-the-difference/)

- [ ] Make sure the text position is under the video and that the video takes the same space it takes in the original page and that it doesn't 'float' to the side of the video.
(Check the screenshot in the comments below for the current state of the clone)

- [ ] The aside container should use the semantic `<aside>` tag, per the project requirements:
aside_requirement

- [ ] The Editor picks aside should contain the editor picks articles with images and links like the original page. (Check the screenshot in the comments below)

- [ ] [OPTIONAL] Add the icons to the top of the article, bottom, and footer using[ font awesome](https://fontawesome.com/) icons or svg icons.

------------

Cheers and Happy coding!👏👏👏

Code review by **Sarah Uryelah Silva Chamorro**

(Feel free to contact me on slack by the handle of  **sarah** if you have any additional question)

**Comment by Honda99** [comment]
Thanks a lot, those comments are very important for this and the next projects.

File: styles.css (hunk 1 above)
Comment (Aduda-Boaz) [reply styles.css]: Thanks a lot for the pointers. We will surely work on them and have a better one for review

File: index.html [hunk 3]
```diff
@@ -379,11 +556,34 @@ <h5>SUBSCRIBE</h5>
...
+                    <li><a href="">NYTCo</a></li>
+                    <li><a href="">Contact Us</a></li>
+                    <li><a href="">Work with us</a></li>
+                    <li><a href="">Election 2020</a></li>
+                    <li><a href="">Cookings</a></li>
+                    <li><a href="">Election 2020</a></li>
+                    <li><a href="">Cookings</a></li>
+                    <li><a href="">Election 2020</a></li>
+                    <li><a href="">Cookings</a></li>
+                    <li><a href="">Subscribtion</a></li>
```
Comment (uche-inyama) [inline index.html]: There are a tags with empty href. Remember to use a hash(#) symbol in between the quotes, if there is no specific file path to include.

File: index.html [hunk 4]
```diff
@@ -379,11 +556,34 @@ <h5>SUBSCRIBE</h5>
...
                         <li><a href="">Games</a></li>
                         <li><a href="">Cookings</a></li>
+                        <li class="list_sep"><a href="">Election 2020</a></li>
+                     
+                        <li><a href="">Cookings</a></li>
+                        <li><a href="">Election 2020</a></li>
+                        <li><a href="">Cookings</a></li>
+                        <li><a href="">Election 2020</a></li>
+                        <li><a href="">Cookings</a></li>
                         <li><a href="">Election 2020</a></li>
```
Comment (uche-inyama) [inline index.html]: There are a tags with empty href. Remember to use a hash(#) symbol in between the quotes, if there is no specific file path to include.

File: index.html [hunk 5]
```diff
@@ -328,6 +471,14 @@ <h5>NEWS</h5>
...
                                 <li><a href="">Politics</a></li>
                                 <li><a href="">Election 2020</a></li>
+                                <li><a href="https://www.nytimes.com/">Home Page</a></li>
+                                <li><a href="https://www.nytimes.com/section/world">World</a></li>
+                                <li><a href="https://www.nytimes.com/section/us"><U class="S">U.S.</U></a></li>
+                                <li><a href="">Politics</a></li>
+                                <li><a href="">Election 2020</a></li>
+                                <li><a href="https://www.nytimes.com/section/us"><U class="S">U.S.</U></a></li>
+                                <li><a href="">Politics</a></li>
+                                <li><a href="">Election 2020</a></li>
```
Comment (uche-inyama) [inline index.html]: There are a tags with empty href. Remember to use a hash(#) symbol in between the quotes, if there is no specific file path to include.

File: index.html [hunk 6]
```diff
@@ -338,6 +489,11 @@ <h5>OPINION</h5>
                                 <li><a href=""><U class="S">U.S.</U></a></li>
                                 <li><a href="">Politics</a></li>
                                 <li><a href="">Election 2020</a></li>
+                                <li><a href="">Home Page</a></li>
+                                <li><a href="">World</a></li>
+                                <li><a href=""><U class="S">U.S.</U></a></li>
+                                <li><a href="">Politics</a></li>
+                                <li><a href="">Election 2020</a></li>
```
Comment (uche-inyama) [inline index.html]: There are a tags with empty href. Remember to use a hash(#) symbol in between the quotes, if there is no specific file path to include.

File: index.html [hunk 7]
```diff
@@ -348,6 +504,13 @@ <h5>ART</h5>
...
+                                <li><a href="">Politics</a></li>
+                                <li><a href="">Election 2020</a></li>
```
Comment (uche-inyama) [inline index.html]: There are a tags with empty href. Remember to use a hash(#) symbol in between the quotes, if there is no specific file path to include.

File: index.html [hunk 8]
```diff
@@ -358,6 +521,15 @@ <h5>LIVING</h5>
...
                                 <li><a href="">Election 2020</a></li>
+                                <li><a href="">Home Page</a></li>
+                                <li><a href="">World</a></li>
+                                <li><a href=""><U class="S">U.S.</U></a></li>
+                                <li><a href="">Politics</a></li>
+                                <li><a href="">Election 2020</a></li>
+                                <li><a href=""><U class="S">U.S.</U></a></li>
+                                <li><a href="">Politics</a></li>
+                                <li><a href="">Election 2020</a></li>
+
```
Comment (uche-inyama) [inline index.html]: There are a tags with empty href. Remember to use a hash(#) symbol in between the quotes, if there is no specific file path to include.

File: index.html [hunk 9]
```diff
@@ -368,6 +540,11 @@ <h5>MORE</h5>
                                 <li><a href=""><U class="S">U.S.</U></a></li>
                                 <li><a href="">Politics</a></li>
                                 <li><a href="">Election 2020</a></li>
+                                <li><a href="">Home Page</a></li>
+                                <li><a href="">World</a></li>
+                                <li><a href=""><U class="S">U.S.</U></a></li>
+                                <li><a href="">Politics</a></li>
+                                <li><a href="">Election 2020</a></li>
```
Comment (uche-inyama) [inline index.html]: There are a tags with empty href. Remember to use a hash(#) symbol in between the quotes, if there is no specific file path to include.

**Review by uche-inyama — CHANGES_REQUESTED** [review]
## Changes Required

### Hi team :wave: 

You have made a great effort to implement most of the changes pointed out by my colleague. However, there is one you missed. 

- [ ] There are  < a > tags with empty hrefs. I have left a comment below for it. I had to repeat my inline comment because there was no way to highlight them all at once.

- [ ] Your web page has a horizontal scroll bar, it should not be there.

#### That's it, after implementing the changes submit another form for a code review.

#### ``` Have a great day ``` :slightly_smiling_face: 

``` Uchechukwu Henry Inyama ```

File: index.html (hunk 3 above)
Comment (Aduda-Boaz) [reply index.html]: Thank you for the pointer, we are surely working on the issues

**Review by uche-inyama — APPROVED** [review]
## Approved

### Hi team :wave: 

Go ahead with the next project.


#### ``` Have a great day ``` :slightly_smiling_face: 

``` Uchechukwu Henry Inyama ```

**Comment by Honda99** [comment]
Thanks a lot, I hope we will do the next project better way this one.

**Pull request merged**